contracts/brush/proc_macros/lib.rs

```diff
@@ -22,41 +22,224 @@ pub fn contract(_attrs: TokenStream, ink_module: TokenStream) -> TokenStream {
     contract::generate(_attrs, ink_module)
 }
 
+/// Marks trait definition to brush as special ink trait definition.
+/// It is the same ink trait definition, but with additional features:
+/// - Allows using super traits
+/// - Allows defining default implementations of methods
+/// - Allows having internal functions(without `#[ink(message)]`)
+/// - Allows calling implementation from trait when overriding(via #[super] + method)
+///
+/// This macro stores definition of the trait in a temporary file during build process.
+/// Based on this definition [`#[brush::contract]`](`macro@crate::contract`)
+/// will generate implementation of this trait. If you defined a default implementation,
+/// [`#[brush::contract]`](`macro@crate::contract`) will copy the default implementation from the trait
+/// and will paste it in impl section. It means that your default implementation must be public
+/// and exported as a part of crate.
+///
+///  ** Note ** You don't need to copy/paste attributes from trait definition, it will be done automatically
+///  ** Note ** Super trait is not used during build process, it is only syntactic sugar for your IDE
+///  ** Note ** Internal methods are not stored in trait, they will be extracted to separate impl section
+/// of your struct, so their implementation also must be public
+///  ** Note ** This macro must be processed before [`#[brush::contract]`](`macro@crate::contract`),
```

Comment:
Do we need this line in doc? I don't think it's necessary for public usage. I guess that mentioning that  `#[brush::trait_definition]` should be defined in `#[brush::contract]` should be enough